### PR TITLE
v0.2.0

### PR DESCRIPTION
**Pull request by MementoRC**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by MementoRC** [comment]
@carterbox This looks like it is almost working but I get this error on windows:
```
Packaging libsecp256k1-1-0.2.0-hcfcfb64_0
ERROR: Glob D:\bld\libsecp256k1-split_1706110970520\_h_env\Library\\bin\\libsecp256k1-1.dll did not match in root_dir D:\bld\libsecp256k1-split_1706110970520\_h_env
```
The library builds correctly and passes its own tests
This was the least-commits PR I made, until I hit windows. It builds on linux and osx. I removed the patching of the relocation, but I am not sure how to test whether it is needed, likely when I use it in `coincurve` we'll know for sure

PS: somehow I did not find how to add you as reviewer
Thx

**Comment by carterbox** [comment]
This error says that when trying to create the `libsecp256k1-{{ so_major_version }}` package, `Library\\bin\\{{ name }}-{{ so_major_version }}.dll` does not exist.

Maybe in this version of libsecp256k1, the Windows DLL is unversioned? i.e. `${PREFIX}/bin/secp256k1-?.dll` doesn't exist, but `${PREFIX}/bin/secp256k1.dll` does?

Just a thought, did you try compiling coincurve v18 with libsecp256k1 v0.4? That would be even easier and [it doesn't look like anything was removed ](https://github.com/bitcoin-core/secp256k1/compare/v0.2.0..v0.4.1#diff-4c52bd5cae65c9519af920228863ca34102abef6414e790498caa9fb05f03bbd) except for some `extern` declarations. So the v0.4.0 might be API backward compatible with v0.2.0, and you wouldn't even need to build this older version of secp256k1.

**Comment by MementoRC** [comment]
@carterbox Thanks for taking a look. I eventually realized that I was not placing the files under `Library` for windows. I forgot to do the same for the include, so next commit should build.

**Comment by carterbox** [comment]
I'll review right now.

**Comment by MementoRC** [comment]
@carterbox Wait, I need one more commit to add all the other builds and re-render. I fugured it's faster and likely less resource consuming to do the debug this way

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,20 +16,32 @@ package:
...
-    - 0001_conda_relocatable.patch
+  sha256: 6cb0fd596e6b1a671f96e9ed7e65a047960def73de024e7b39f45a78ab4fc8df
+  # patches:
+  #   # Use the /opt/anaconda1anaconda2anaconda3 prefix for the conda build relocatable
+  #   # https://docs.conda.io/projects/conda-build/en/stable/resources/make-relocatable.html
+  #   - 0001_conda_relocatable.patch
 
 build:
   number: 0
+  skip: true  # [not win and not py==3.9]
```
Comment (carterbox) [inline recipe/meta.yaml]:
```suggestion
  skip: true  # [py != 39]
```

**Review by carterbox — CHANGES_REQUESTED** [review]
You want to merge this into a new `0.2.x` branch instead of the `main` branch otherwise, you will have to revert these changes for the next libsecp256k1 release (0.5.0 or whatever it is).

Please enable all of the builds and rerender, so I can see the log output for all variants.

**Comment by MementoRC** [comment]
@carterbox Hopefully, it won't suddenly break o.0

Right, I did and yes, I intend to create a new branch and merge to it ... is there a way to do this automatically from Pycharm?

File: recipe/meta.yaml (hunk 1 above)
Comment (MementoRC) [reply recipe/meta.yaml]: This was only to limit the amount of builds, I don't need any `skip`

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by carterbox** [comment]
> Right, I did and yes, I intend to create a new branch and merge to it ... is there a way to do this automatically from Pycharm?

You can do it from the GitHub website. Just go to the branches page for this feedstock and then click the "New branch" button. Once the new branch is created, you can [change the target of this PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) from the top of this PR.

**Review by carterbox — APPROVED** [review]

**Comment by MementoRC** [comment]
@carterbox Thank you for your help. I am going to merge since I only changed the base branch

**Pull request merged**